R/combine_checkboxes.R

```diff
@@ -0,0 +1,192 @@
+#' values under. Required.
+#' @param multi_value_label A string specifying the value to be used when multiple
+#' checkbox fields are selected. Default "Multiple".
+#' @param values_fill Value to use when no checkboxes are selected. Default `NA`.
+#' @param raw_or_label Either 'raw' or 'label' to specify whether to use raw coded
+#' values or labels for the options. Default 'label'.
+#' @param keep Logical indicating whether to keep the original checkbox fields in
+#' the output. Default `TRUE`.
+#'
+#' @return A modified supertibble.
+#'
+#' @examples
+#' \dontrun{
+#' supertbl <- read_redcap(redcap_uri, token)
+#' combined_tbl <- combine_checkboxes(
+#'   supertbl = supertbl,
+#'   tbl = "demographics",
+#'   cols = starts_with("race"),
+#'   values_to = "race_combined",
+#'   multi_value_label = "Multiple",
+#'   values_fill = NA
+#' )
+#' }
+#'
+#' @export
+
+combine_checkboxes <- function(supertbl,
+                               tbl,
+                               cols,
+                               values_to,
+                               multi_value_label = "Multiple",
+                               values_fill = NA,
+                               raw_or_label = "label",
+                               keep = TRUE) {
+
+  # Check args ---
+  check_arg_is_supertbl(supertbl, req_cols = c("redcap_data", "redcap_metadata"))
+  check_arg_is_character(tbl, len = 1, any.missing = FALSE)
+  check_arg_is_character(values_to, len = 1, any.missing = FALSE)
+  check_arg_is_character(multi_value_label, len = 1, any.missing = TRUE)
+  check_arg_is_character(values_fill, len = 1, any.missing = TRUE)
+  check_arg_choices(raw_or_label, choices = c("label", "raw"))
+  check_arg_is_logical(keep, len = 1, any.missing = FALSE)
+
+  # Save user cols to enquosure
+  cols_exp <- enquo(cols)
+
+  # Extract tbl from supertbl
+  data_tbl <- supertbl %>%
+    extract_tibble(tbl)
+
+  # Get field names from cols_exp, check that fields exist
+  field_names <- names(eval_select(cols_exp, data = data_tbl))
+  check_fields_exist(fields = field_names, expr = cols_exp)
+
+  # Assume the first instrument in the metadata contains IDs
+  # REDCap enforces this constraints, we reflect this in read_redcap -> get_field_to_drop
+  record_id_field <- supertbl$redcap_metadata[supertbl$redcap_form_name == tbl][[1]]$field_name[1]
+
+  # Combine record identifier with remaining possible project identifiers
+  instrument_identifiers <- c(
+    record_id_field,
+    "redcap_form_instance",
+    "redcap_form_name",
+    "redcap_event",
+    "redcap_event_instance"
+  )
+
+  # Define values_to as the count of TRUEs/1s for the given checkbox field
+  # Assign TRUE if multiple selections made, and FALSE if one or zero made
+  data_tbl_mod <- data_tbl %>%
+    select(any_of(instrument_identifiers), !!!eval_select(cols_exp, data_tbl)) %>%
+    mutate(
+      !!values_to := case_when(
+        rowSums(select(., eval_tidy(cols_exp))) > 1 ~ TRUE,
+        TRUE ~ FALSE
+      )
+    )
```

Comment:
I kept the identifiers specifically for the join at the end, as things stood without them you wouldn't be able to combine the modified data back with the original. When I was making this it helped me more visually to section off the data for only the things I was doing modifications to but I suppose the `select()` can be dropped in favor of `mutate()` with `across()`?

I'll think on this more, I agree it can be condensed and cleaned up. In making sure the output was what we want it made the most sense to me to start with making a modified table that would get re-combined at the end.